### ISSUE TITLE
Steam: No sound in many games

### ISSUE THREAD

**Issue by baracoder** [NixOS/nixpkgs#15418]
## Issue description

When starting games in steam, many do not have any sound. They do not appear in the pulseaudio volume mixer.

So far no audio in: Armikrog, Deadnaut, Read Only Memories, TIS-100

This is the log of starting Factorio, as it actually does some log output.

```
...
   0.112 Initialised OpenGL:[0] GeForce GTX 570/PCIe/SSE2; driver: 4.5.0 NVIDIA 361.42
ALSA lib conf.c:3750:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
ALSA lib pcm.c:2266:(snd_pcm_open_noupdate) Unknown PCM default
   0.116 Warning Display.cpp:36: Installing audio failed retrying...
ALSA lib conf.c:3750:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
ALSA lib pcm.c:2266:(snd_pcm_open_noupdate) Unknown PCM default
   0.617 Warning Display.cpp:36: Installing audio failed retrying...
ALSA lib conf.c:3750:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
ALSA lib pcm.c:2266:(snd_pcm_open_noupdate) Unknown PCM default
   1.117 Warning Display.cpp:36: Installing audio failed retrying...
ALSA lib conf.c:3750:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
ALSA lib pcm.c:2266:(snd_pcm_open_noupdate) Unknown PCM default
   1.617 Warning Display.cpp:36: Installing audio failed retrying...
ALSA lib conf.c:3750:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
ALSA lib pcm.c:2266:(snd_pcm_open_noupdate) Unknown PCM default
   2.117 Warning Display.cpp:36: Installing audio failed
   2.656 Loading mod core 0.0.0 (data.lua)
...
```

Also Prison Architect crashes
```
Initialising SDL Audio:
ALSA lib conf.c:3750:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
ALSA lib pcm.c:2266:(snd_pcm_open_noupdate) Unknown PCM default
FATAL ERROR : Failed to open audio output device: "ALSA: Couldn't open audio device: No such file or directory"
Failed to open audio output device: "ALSA: Couldn't open audio device: No such file or directory"SDL Version: Compiled against 2.0.3, running with 2.0.3
/home/bara/.local/share/Steam/steamapps/common/Prison Architect/PrisonArchitect: line 59:  2953 Aborted                 "$base"/PrisonArchitect.x86_64 "$@"
Game removed: AppID 233450 "Prison Architect", ProcID 2946 
No cached sticky mapping in ActivateActionSet.

```

Infinifactory crashes to, but has no useful logs.

I have pulseaudio installed and enabled support for 32bit:

```
  hardware.pulseaudio = {
    enable = true;
    support32Bit = true;
    package = pkgs.pulseaudioFull;
  };

```

### Steps to reproduce

1. install steam (on unstable)
2. Run any of the games

## Technical details

* System: NixOS 16.09pre83100.25e3c09 (Flounder)
* Nix version:  nix-env (Nix) 1.11.2
* Nixpkgs version: "16.09pre83100.25e3c09"

**Comment by abbradar** [comment]
Fixed in 229e544. Thanks for reporting!

**Comment by baracoder** [comment]
Confirmed, works now.
Thanks!
